### DISCUSSION TITLE
template with many parameters

### DISCUSSION THREAD

**Discussion by fadhil-riyanto** [ahmadfaizalbh/Chatbot#62] [Q&A] [answered]
Hi all, I have 1 question
i have the string "1 usd berapa rupiah"
and i try getting "1", "usd", and "rupiah"

my template :
```
{% block %}
    {% client %} (?P<nomoruang>.*) (?P<mata1>.*) (berapa) (?P<mata2>.*){% endclient %}
    {% response %}{% call f_duid_converter: %nomoruang %mata1 %mata2 %}{% endresponse %}
{% endblock %}
```

question : how do i pass the result to the nomoruang, mata1, mata2 
in python function? i try in [Example.py line 9](https://github.com/ahmadfaizalbh/Chatbot/blob/07941eb683a13ed0fed024cd8ee7ca5c3f703afb/examples/Example.py#L9) 
and get error "TypeError: f_duid_converter() missing 2 required positional arguments: 'mata1' and 'mata2'"

my python code:
```
@register_call("f_duid_converter")
def f_duid_converter(session, nomoruang, mata1, mata2):
    from currency_converter import CurrencyConverter
    c = CurrencyConverter()
    hasliuang = c.convert(nomoruang, mata1, mata2)
    return hasliuang
```

**Comment by ahmadfaizalbh** [comment, marked as answer]
Please use as below
Template:
```
{% block %}
    {% client %}(?P<nomoruang>.*) (?P<mata1>.*) (berapa) (?P<mata2>.*){% endclient %}
    {% response %}{% call f_duid_converter: %nomoruang %}{% endresponse %}
{% endblock %}
```

python code:
```python
@register_call("f_duid_converter")
def f_duid_converter(session, nomoruang):
    from currency_converter import CurrencyConverter
    c = CurrencyConverter()
    data = session.attr['match'].groupdict()
    hasliuang = c.convert(nomoruang, data['mata1'], data['mata2'])
    return hasliuang
```

As of now only one parameter can be passed to function from template but you can get all match group data as dict using ` session.attr['match'].groupdict()`.
 we are working on multi-parameter support which will be available in future versions .

**Reply by fadhil-riyanto** [reply]
Wow work... thank
